tests/units/units.cpp

```diff
@@ -891,3 +891,164 @@ TEST(Units, complicatedMultiplicationFactorUnits)
     // Incompatible units but return a scaling factor.
     EXPECT_EQ(1e-07, libcellml::Units::scalingFactor(incredible_pile_of_square_apples, bunch_of_bananas));
 }
+
+TEST(Units, checkingOwningModelBothUnits)
+{
+    libcellml::ModelPtr model = libcellml::Model::create();
+
+    libcellml::UnitsPtr u = libcellml::Units::create();
+    u->setName("u");
+
+    libcellml::UnitsPtr u1 = libcellml::Units::create();
+    u1->setName("u1");
+    u1->addUnit("u", "milli", 2.0, 1000.0); //m^2
+    u1->addUnit("dimensionless");
+
+    libcellml::UnitsPtr u2 = libcellml::Units::create();
+    u2->setName("u2");
+    u2->addUnit("u", "kilo", 2.0, 0.001); // standard, exponent.
+    u2->addUnit("dimensionless");
+
+    libcellml::UnitsPtr u3 = libcellml::Units::create();
+    u3->setName("u3");
+    u3->addUnit("u", "kilo", 4.0, 0.001); // standard, exponent.
+
+    model->addUnits(u1);
+    model->addUnits(u2);
+    model->addUnits(u3);
```

Comment:
Do we have a test where `u3` here is not added to the model but has a reference to an units that is in the model?

Reply:
We don't currently, but I can see how perhaps that would give something we wouldn't expect - I think at the moment the `owningModel` call means that such a combination would work, but considering that the units should all be part of the same model it might not make sense for it to work.